src/cupbearer/detectors/anomaly_detector.py

```diff
@@ -19,6 +19,9 @@
 
 
 class AnomalyDetector(ABC):
+    should_train_on_clean_data: bool = False
+    should_train_on_poisoned_data: bool = False
```

Comment:
Do we need two different fields here? I think it should always be only one or the other. Also, I'd probably call it "untrusted" instead of "poisoned" because (1) outside of backdoors, "poisoned" doesn't really apply, and (2) not all of it will be poisoned, it will be some mix. In general our terminology is admittedly pretty inconsistent, I think "trusted" for clean, "anomalous" for anomalous/poisoned, and "untrusted" for mix might be best, staying close to the RR paper. "Clean" also seems fine

Reply:
So to be clear, I'm suggesting having e.g. only a `should_train_on_trusted_data` with no default (so maybe an abstractproperty?). Then whereever we'd use should_train_on_poisoned_data, we just invert that instead

Reply:
Spectre might work for both cases after some adaption. The paper describes a method for cleaning a poisoned dataset, but the metric is described in relation to clean data.

Though, as it is now it could just as well be a dichotomy.

Reply:
Oh I see, I'd forgotten that I'd said a while ago to change SPECTRE and just assume access to clean data. Arguably we shouldn't call this "SPECTRE" given that the robust estimation procedure is a key part of what the paper focuses on. I'd suggest:
* We call the current `SpectreDetector` something like `QUESpectralDetector` (I think in your implementation, the main difference to the spectral detector is QUE?). That detector will only support clean data
* If we ever want the robust estimation version of SPECTRE, we make a new class that only supports untrusted data (but I don't think this is high priority)
* More generally, I think it's probably reasonable for detectors to only work with one version. If a detector supports multiple versions with slightly different algorithms, then we should either have two classes, or a config option to the detector (and `should_train_on_trusted_data` could be a property that uses that config option to decide). I think the only case where we wouldn't want this if there was some algorithm that magically worked with both cases, without any modifications (personally I lean towards just ignoring that for now, but haven't thought too much about whether there might be cases like this)

Reply:
Sounds good I'll:
- [ ] do this tomorrow

Reply:
I changed the name Spectre -> QuantumEntropy and spectre -> que. As it is now I find it conceptually more easy to think of it like the MahalanobisDetector but with QUE-score instead of L2 norm.